src/c2goto/library/pthread_lib.c

```diff
@@ -45,6 +49,68 @@ pthread_t __ESBMC_get_thread_id(void);
 void __ESBMC_really_atomic_begin(void);
 void __ESBMC_really_atomic_end(void);
 
+/************************** Linked List Implementation **************************/
+
+typedef struct thread_key
+{
+  void *thread;
+  void *key;
+  void *value;
+  struct list *next;
+} __ESBMC_thread_key;
+
+__ESBMC_thread_key *head = NULL;
+__ESBMC_thread_key *mylist;
+
+int insert_key_value(__ESBMC_thread_key *l, void *key, void *value)
+{
+  l = (__ESBMC_thread_key *)malloc(sizeof(__ESBMC_thread_key *));
+  if(l == NULL)
+  {
+    return -1;
+  }
+  l->thread = __ESBMC_get_thread_id();
+  l->key = key;
+  l->value = value;
+  l->next = (head == NULL) ? NULL : head;
+  head = l;
+  return 0;
+}
+
+void *search_key(__ESBMC_thread_key *l)
+{
+  l = head;
+  while(l != NULL && l->thread != __ESBMC_get_thread_id())
+  {
```

Comment:
No need for braces in one line `while`s.

Reply:
Fixed.